src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -561,6 +561,8 @@ private BoundStatement BindLocalFunctionStatement(LocalFunctionStatementSyntax n
             var hasErrors = localSymbol.ScopeBinder
                 .ValidateDeclarationNameConflictsInScope(localSymbol, diagnostics);
 
+            ReportFieldContextualKeywordConflictIfAny(localSymbol, node, node.Identifier, diagnostics);
```

Comment:
I was curious if there was any more "central" location to do this check, like where local symbols are gathered up into a scope, or some such. It looks like probably not. Places like `LocalScopeBinder.MakeLocal` are not really designed for reporting diagnostics.

Reply:
I haven't found a central location to catch these cases.

Reply:
Could we perhaps catch all parameters in `ParameterHelpers.MakeParameters`?

Reply:
`ParameterHelpers.MakeParameters()` will not work as is because that method is not used for lambda parameters (`LambdaSymbol.MakeParameters()` is used instead), and the method does not include a `Binder` which is used for reporting. I'll leave this as is.